src/evaluate/evaluator/question_answering.py

```diff
@@ -0,0 +1,191 @@
+                Argument can be used to overwrite a default tokenizer if `model_or_pipeline` represents a model for
+                which we build a pipeline. If `model_or_pipeline` is `None` or a pre-initialized pipeline, we ignore
+                this argument.
+            strategy: (`Literal["simple", "bootstrap"]`, defaults to "simple"):
+                specifies the evaluation strategy. Possible values are:
+                - `"simple"` - we evaluate the metric and return the scores.
+                - `"bootstrap"` - on top of computing the metric scores, we calculate the confidence interval for each
+                of the returned metric keys, using `scipy`'s `bootstrap` method
+                https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.bootstrap.html.
+            confidence_level (`float`, defaults to `0.95`):
+                The `confidence_level` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            n_resamples (`int`, defaults to `9999`):
+                The `n_resamples` value passed to `bootstrap` if `"bootstrap"` strategy is chosen.
+            random_state (`int`, *optional*, defaults to `None`):
+                The `random_state` value passed to `bootstrap` if `"bootstrap"` strategy is chosen. Useful for
+                debugging.
+            question_column (`str`, defaults to `"question"`):
+                the name of the column containing the question in the dataset specified by `data`.
+            context_column (`str`, defaults to `"context"`):
+                the name of the column containing the context in the dataset specified by `data`.
+            id_column (`str`, defaults to `"id"`):
+                the name of the column cointaing the identification field of the question and answer pair in the
+                dataset specified by `data`.
+            label_column (`str`, defaults to `"label"`):
+                the name of the column containing the answers in the dataset specified by `data`.
+        Return:
+            A `Dict`. The keys represent metric keys calculated for the `metric` spefied in function arguments. For the
+            `"simple"` strategy, the value is the metric score. For the `"bootstrap"` strategy, the value is a `Dict`
+            containing the score, the confidence interval and the standard error calculated for each metric key.
+        Examples:
+        ```python
+        >>> from evaluate import evaluator
+        >>> from datasets import Dataset, load_dataset
+        >>> e = evaluator("question-answering")
+        >>> data =  Dataset.from_dict(load_dataset("squad")["validation"][:2])
+        >>> results = e.compute(
+        >>>     model_or_pipeline="mrm8488/bert-tiny-finetuned-squadv2",
+        >>>     data=data,
+        >>>     metric="squad",
+        >>>     question_column="question",
+        >>>     context_column="context",
```

Comment:
Yes they are the default! I just followed upon `TextClassificationEvaluator` example which also had the example 

```
        >>> results = e.compute(
        >>>     model_or_pipeline="huggingface/prunebert-base-uncased-6-finepruned-w-distil-mnli",
        >>>     data=data,
        >>>     metric="accuracy",
        >>>     input_column="text",
        >>>     label_column="label",
        >>>     label_mapping={"LABEL_0": 0.0, "LABEL_1": 1.0},
        >>>     strategy="bootstrap",
        >>>     n_resamples=10,
        >>>     random_state=0
        >>> )
```

i.e. `input_column` and `label_column` with the defaults. Do you think we should remove them from all examples? In any case, if the user runs with the default and the column is missing we have nice handling and nice error raised, so I am in favor of removing the default args. Do you agree?

Reply:
Yes, let's keep the examples to a minimal, otherwise it might look a bit overwhelming. We can still explain in detail how to use the extra kwargs in the docs.